### ISSUE TITLE
Cannot set default button classes

### ISSUE THREAD

**Issue by matthew-inamdar** [t4t5/sweetalert#785]
I've read the documentation and cannot figure this out.

How does one set the default button class for the confirm and cancel buttons?

The following does not work:

```js
swal.setDefaults({
    buttons: {
        cancel: {
            className: 'btn btn-light'
        },
        confirm: {
            className: 'btn btn-success'
        }
    }
});
```

**Comment by girishpatil** [comment]
Css 

`.red-bg{
  background :red
}
.blue-bg{
  background :blue
}`
JS

```
swal({
  buttons: {
    dance : {
      text : "Button 1",
      className : 'red-bg'
    },
    confirm: {
      text : 'Button 2',
      className : 'blue-bg'
    }
  }
});
```

Know more here https://sweetalert.js.org/docs/#buttons

**Comment by matthew-inamdar** [comment]
@girishpatil - this doesn't solve the problem, unfortunately. This requires you to set the button classes on every call to `swal()` (which is what I'm currently doing).

This results in duplicated code.

As the title says, I cannot set the **default** button class.

**Comment by girishpatil** [comment]
@matthew-inamdar yeah sorry for that .. although I couln't find a way of assigning external classes but digging into the code found out that there are three types of default buttons. 
You can use their default class names to style them
Or you can even extend `@extend` your own class (btn btn-left as mentined above) into these below classes

`/*  style overall buttons */
.swal-button{
  border: 1px dashed #333;
// @extend .yourclass
}
/*  style cancel buttons */
.swal-button--cancel{
  color: #333;  
}
.swal-button--confirm{
  color: green 
}
.swal-button--danger{
  color: red;
}`
hope it helps in some way
you can see the demo here https://codepen.io/girishpatil/pen/wpERBB

**Comment by matthew-inamdar** [comment]
> Or you can even extend `@extend` your own class (btn btn-left as mentined above) into these below classes

This is an interesting solution. Not as clean as I would have liked, but may do the trick.

I would like this issue to remain open as hopefully there is still an actual way with the default config/

If this is currently not possible (for someone to confirm) - I may open a PR if I get the time.

**Comment by t4t5** [comment]
I agree that this should work, so yeah, this could be classified as a bug.

I might tackle it sometime in the future, but if you have the time to submit a PR, that would be great!

**Comment by matthew-inamdar** [comment]
Thanks for getting back to me @t4t5 - I'll find some time this week hopefully

**Comment by jsainthilaire** [comment]
@t4t5 @matthew-inamdar I'm looking for something to contribute between today and tomorrow, I can take over this, is that ok?

**Comment by Riya123-code** [comment]
can i work on this bug?
